### ISSUE TITLE
New .csproj format causes FabActUtil to fail

### ISSUE THREAD

**Issue by amanbha** [Microsoft/service-fabric-services-and-actors-dotnet#79]
Opening it to track as reported by @VikSab in comments of issue #72

This issue seems to be related to using the new csproj format - even though the target framework is 4.6.1 - if the project is moved to new csproj format - its being detected as netstandard/netcoreapp

This issue is causing our build to break - we are in .net 4.6.1 - and want to use latest version of the Service Fabric - But the FabActUtil.exe is failing on build
We are unable to define a custom serialization attribute for our actors.
for some reason this is being resolved to using RemotingListener.V1Listener causing the FabActUtil to load a ActorRemotingProviderAttribute with 2 additional abstract methods that need to be implemented - but arent showing up when we try to create derived type of ActorRemotingProviderAttribute
[assembly: CustomActorRemotingProvider(RemotingListener = RemotingListener.V2Listener, RemotingClient = RemotingClient.V2Client)]
[assembly: CustomServiceRemotingProvider(RemotingListener = RemotingListener.V2Listener, RemotingClient = RemotingClient.V2Client)]
Here is the failure
System.TypeLoadException: Method 'CreateServiceRemotingListener' in type 'Dover.DataServices.ServiceFabric.Services.Remoting.Json.JsonActorRemotingProviderAttribute' from assembly 'ServiceFabric.JsonRemoting, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
at System.ModuleHandle.ResolveType(RuntimeModule module, Int32 typeToken, IntPtr* typeInstArgs, Int32 typeInstCount, IntPtr* methodInstArgs, Int32 methodInstCount, ObjectHandleOnStack type)
at System.ModuleHandle.ResolveTypeHandleInternal(RuntimeModule module, Int32 typeToken, RuntimeTypeHandle[] typeInstantiationContext, RuntimeTypeHandle[] methodInstantiationContext)
at System.Reflection.RuntimeModule.ResolveType(Int32 metadataToken, Type[] genericTypeArguments, Type[] genericMethodArguments)
at System.Reflection.CustomAttribute.FilterCustomAttributeRecord(CustomAttributeRecord caRecord, MetadataImport scope, Assembly& lastAptcaOkAssembly, RuntimeModule decoratedModule, MetadataToken decoratedToken, RuntimeType attributeFilterType, Boolean mustBeInheritable, Object[] attributes, IList derivedAttributes, RuntimeType& attributeType, IRuntimeMethodInfo& ctor, Boolean& ctorHasParameters, Boolean& isVarArg)
at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeModule decoratedModule, Int32 decoratedMetadataToken, Int32 pcaCount, RuntimeType attributeFilterType, Boolean mustBeInheritable, IList derivedAttributes, Boolean isDecoratedTargetSecurityTransparent)
at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeAssembly assembly, RuntimeType caType)
at System.Attribute.GetCustomAttributes(Assembly element, Type attributeType, Boolean inherit)
at System.Attribute.GetCustomAttribute(Assembly element, Type attributeType, Boolean inherit)
at System.Reflection.CustomAttributeExtensions.GetCustomAttribute[T](Assembly element)
at Microsoft.ServiceFabric.Actors.Remoting.ActorRemotingProviderAttribute.GetProvider(IEnumerable1 types) at Microsoft.ServiceFabric.Actors.Runtime.ActorTypeInformation.Get(Type actorType) at FabActUtil.Tool.LoadActors(Assembly inputAssembly, IList1 actorFilters, IList`1 actorTypes)
at FabActUtil.Tool.LoadActors(ToolContext context)
at FabActUtil.Tool.ProcessInput(ToolContext context)
at FabActUtil.Tool.Run(ToolArguments arguments)
at FabActUtil.Program.Main(String[] args)

**Comment by amanbha** [comment]
@VikSab As per the comments in #72 you are unblocked by using the old .csproj. That's great!

The VS templates are still generating older .csproj for full .net framework actor projects, so please don't change them until its supported for Actors project. I cannot confirm ETA for supporting the new .csproj at this point as there are other high work items we are working on, but will get to it as soon as possible. Will create a new issue to track this new .csproj format failure you are seeing.

**Comment by VikSab** [comment]
@amanbha  - the actors project was in the old format for precisely the reason that you have given

our custom serialization was in another project that had been moved to the new format. 

The FabActUtil was still not able to load the Actor assembly.

The other project wasnt even able to see V1 implementations without reverting to the Old csproj format

**Comment by amanbha** [comment]
@VikSab To confirm the other project containing custom serialization, it was also a full .net framework project, correct? Would it be possible to share a repro?

**Comment by VikSab** [comment]
@amanbha - yes it is also a net461 project ... converted the project over to the new format - as stated - we are able to see the non-dotnet core implementations just by reverting to the old format - as soon as we move to new format we cant see the implementations either.. (the project wont even build) - it should be a simple repro at your end -- if not i can work on one that will work for you guys.

**Comment by amanbha** [comment]
@VikSab Thanks for explaining, so it seems like with new .csproj, even though its full .net framework, it referenced binaries for net core, correct? If that's the case I will try creating repro at my end.

**Comment by VikSab** [comment]
@amanbha - i would assume that's correct - in the new format there doesnt seem to be a way to give the hint path for nuget

**Comment by VikSab** [comment]
@amanbha  - let me see if what is the case if i specify the target version in the nuget reference

**Comment by VikSab** [comment]
@amanbha - even adding reference in new csproj format with targetFramework specified - still seems to pick the .netcore version    <PackageReference Include="Microsoft.ServiceFabric.Actors" Version="3.0.456" targetFramework="net461" />

**Comment by amanbha** [comment]
@VikSab Thanks for confirming, will look at it.

**Comment by amanbha** [comment]
related to https://github.com/Azure/service-fabric-services-and-actors-dotnet/issues/94

**Comment by amanbha** [comment]
Fix available in Microsoft.ServiceFabric.Actors nuget package version >= 3.1.283
